src/components/KPI/KPI.jsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+export default function KPI({ qlikApp, objectId, title, color }) {
+  // Define states
+  const [kpiObject, setKpiObject] = useState();
+
+  // Get the kpiObject, using the props provided and set to state
+  useEffect(() => {
+    // Get the kpiObject, which is an instance of a qlik object & set to state
+    const getSetObjectDetails = async () => {
+      // Get the specified object from the instance of the specific app
+      const qObject = await qlikApp.getObject(objectId);
+      setKpiObject(qObject);
+      // console.log('qObject', qObject);
+    };
+    // Only call the getSetObject when: kpiObject doesn't exist, qlikApp does exist and objectId does exist
+    if (!kpiObject && qlikApp && objectId) {
+      getSetObjectDetails();
+    }
+  }, [kpiObject, qlikApp, objectId]);
+
+  // Get the relevant kpiData from the layout of the kpiObject and set to state
+  const [kpiData, setKpiData] = useState(undefined);
+  useEffect(() => {
+    const getKpiData = async () => {
+      const qPath = '/qHyperCubeDef';
+      const qPages = [
+        {
+          qLeft: 0,
+          qTop: 0,
+          qWidth: 2, // layoutObject['qHyperCube']['qSize']['qcx']
+          qHeight: 2, // layoutObject['qHyperCube']['qSize']['qcy']
+        },
+      ];
+
+      const layout = await kpiObject.getHyperCubeData(qPath, qPages);
+      console.log('layout', layout);
+      const newKpiData = {
+        value: Math.round(layout[0].qMatrix[0][0].qNum / 10000000),
+        percent: layout[0].qMatrix[0][1].qNum * 100,
+      };
+      setKpiData(newKpiData);
+    };
+    if (kpiObject && !kpiData) {
+      getKpiData();
+    }
+  }, [kpiData, kpiObject]);
+  let arrow;
+  // Determine the direction of the arrow symbol, based on percentage value
+  if (kpiData?.percent < 0) {
+    arrow = 'down';
+  } else if (kpiData?.percent > 0) {
+    arrow = 'up';
+  } else {
+    arrow = '';
+  }
+  return (
+    <Section color={color}>
+      <h5>£ {kpiData?.value.toLocaleString()} m</h5>
+      <Percent direction={arrow}>
+        <h6>{title}</h6>
+        <span className="percent">
+          <i className={`fa-solid fa-arrow-${arrow}`} />
+          {kpiData?.percent.toFixed(2) + '%'}
+        </span>
+      </Percent>
+    </Section>
+  );
+}
+const Section = styled.div`
+  justify-content: center;
+  padding: 1rem 1rem 0rem 1rem;
+  border-radius: 1rem;
+  color: black;
+  justify-content: space-evenly;
+  height: 5rem;
+  width: max-content;
+  align-items: center;
+  background-color: ${(props) => props.color};
+  transition: 0.5s ease-in-out;
+  box-shadow: 0 0 20px rgba(0, 0, 0, 0.05), 0 0px 40px rgba(0, 0, 0, 0.08);
+
+  &:hover {
+    background-color: #d4e0ff;
+    color: black;
+    cursor: pointer;
+  }
+  h5 {
+    margin-bottom: 0;
+    margin-top: 0;
+  }
+`;
+const Percent = styled.div`
+  .percent {
+    color: ${(props) =>
```

Comment:
FYI:
Not a big fan of nested ternaries, here it's not too bad, but I've seen people to crazy nested ternaries to render different JSX and it gets really confusing.
Also, on some projects this rule is applied:
https://eslint.org/docs/rules/no-nested-ternary

One alternative would be to (there are other ways you could do an if else statement):
```
// In the styled component
const Percent = styled.li`
  color: ${(props) => props.color || 'grey' }); // ****
  ...

**** This seems a bit weird cos we're returning 'darkslategrey' as a default, but just as a failsafe, 
it's always good to add a backup in case the function gets broken for whatever reason.

// In the Body of the component
const getCorrectColor = (arrow) => {
  switch(arrow) {
    case 'up':
      return 'green'
    case 'down':
      return 'red'
    default:
      return 'grey'
  }
}
  
// In the JSX
<Percent color={getCorrectColor(arrow)}>
```